etcd-manager/pkg/volumes/aws/volumes.go

```diff
@@ -104,8 +104,7 @@ func NewAWSVolumes(clusterName string, volumeTags []string, nameTag string) (*AW
 	}
 
 	a.localIP, err = a.metadata.GetMetadata("ipv6")
-	awsErr, isAWSErr := err.(awserr.RequestFailure)
-	if !isAWSErr || awsErr.StatusCode() != 404 {
+	if !isNotFoundError(err) {
```

Comment:
Also, maybe `a.ec2 = ec2.New(s, config.WithRegion(region))` can be put before this bloc to not duplicate the code before return.

```suggestion
	if err == nil {
		a.ec2 = ec2.New(s, config.WithRegion(region))
 		return a, nil
 	}
 	
	awsErr, isAWSErr := err.(awserr.RequestFailure)
	if !isAWSErr || awsErr.StatusCode() != 404 {
```

Reply:
Won't this duplicate the logic? This is doing one `return a` for ipv6 and then a `return a` for ipv4.

Reply:
I assume that if we have ipv6, we want to return that one. As is in the PR, it will return ipv4 for dual-stack too.

Reply:
You are absolutely correct. I missed that.